apis/tempo/v1alpha1/microservices_types.go

```diff
@@ -162,9 +162,34 @@ const (
 	Dynamic ModeType = "dynamic"
 )
 
+// MultitenancySpec defines if gateway, qurier and db multitenancy support should
+// be enabled and which header is used to identify tenants.
+type MultitenancySpec struct {
+	// Enable multi tenancy support in tempo. Default is false.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default:=false
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Multitenancy enabled"
+	Enabled bool `json:"enabeld"`
+
+	// Header used to identify tenant.
+	// +optional
+	// +kubebuilder:example="X-Scope-OrgID"
+	Header string `json:"tempoQueryVersion,omitempty"`
```

Comment:
What is the use case for specifying the header? 

as far as I know tempo upstream supports only `X-Scope-OrgID`

Reply:
Iam not sure. Is it not configurable in tempo and jaeger?
https://github.com/jaegertracing/jaeger/blob/913ea321c9e1149871f7561bb69c497bd7316545/pkg/tenancy/flags.go#L35

nvm, ive no strong opinion on it. Its now replaced by a constant value.

Reply:
right, but it's not configurable in tempo. So without the gateway, the config does not make sense.